sos/plugins/unpackaged.py

```diff
@@ -0,0 +1,86 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, RedHatPlugin
+import os
+import re
+
+
+class Unpackaged(Plugin, RedHatPlugin):
+    '''
+    Collects a list of files that are not handled by the package
+    manager
+    '''
+
+    def setup(self):
+
+        def all_files_rpm():
+            """
+            It returns a list of files known by the package manager
+            """
+            return self.get_command_output('rpm -qal')['output'].splitlines()
+
+        def get_env_path_list():
+            """
+            It returns a list of directories contained in the $PATH var
+            """
+            return os.environ['PATH'].split(':')
+
+        def check_usrmove():
+            """
+            It checks if '/bin' and '/sbin' are symlink to '/usr/[s]bin'.
+            It returns True if both paths are symlinks.
+            """
+            return os.path.islink('/bin') and os.path.islink('/sbin')
+
+        def mangle_usrmove(files):
+            """
+            files: list
+
+            If we are on a post-usrmove system, all files will be in '/usr/[s]bin'.
+            This will substitute all the /[s]bin references in the 'files' list
+            with '/usr/[s]bin'
+            """
+            return [re.sub(r'(^)(/s?bin)', r'\1/usr\2', f) for f in files]
+
+        def all_files_system(path, exclude=None):
+            """
+            path: string
+            exclude: list
+
+            It returns a list of all files installed on a system.
+            It excludes all the files in the directories listed in 'exclude'
+            """
+            file_list = []
+
+            print("path: {}".format(path))
+            for root, dirs, files in os.walk(path, topdown=True):
+                if exclude:
+                    dirs[:] = [d for d in dirs if dirs not in exclude]
+                for name in files:
+                    file_list.append(os.path.join(root, name))
+                for name in dirs:
+                    file_list.append(os.path.join(root, name))
+
+            return file_list
+
+        all_fsystem = []
+        all_frpm = set(mangle_usrmove(all_files_rpm()))
```

Comment:
I'd suggest something like a policy or package manager method `mangle_package_path()` - in the default case it simply returns the string unchanged, but for post-UsrMove distros it can implement the necessary path rewriting.